sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -62,7 +75,10 @@ object IntegerLiteral {
   }
 }
 
-case class Literal(value: Any, dataType: DataType) extends LeafExpression {
+case class Literal (var value: Any, dataType: DataType) extends LeafExpression {
```

Comment:
I think you are right. Can we submit a separate PR to avoid using the constructor directly? Then we can merge this one.